core/src/main/java/org/mule/runtime/core/exception/TemplateOnErrorHandler.java

```diff
@@ -196,30 +196,18 @@ public void setWhen(String when) {
 
   @Override
   public boolean accept(MuleEvent event) {
-    return acceptsAll() || acceptsEvent(event) || acceptsErrorType(event)
+    return acceptsAll() || acceptsErrorType(event)
         || (when != null && muleContext.getExpressionManager().evaluateBoolean(when, event, flowConstruct));
   }
 
   private boolean acceptsErrorType(MuleEvent event) {
     //For now, match by representation, since we can't assign the exact type when parsing
-    return event.getError().getErrorType().getStringRepresentation().equals(errorType.getStringRepresentation());
-  }
-
-  /**
-   * Determines if the exception strategy should process or not a message inside a choice exception strategy.
-   * <p>
-   * Useful for exception strategies which ALWAYS must accept certain types of events despite when condition is not true.
-   *
-   * @param event   The MuleEvent being processed
-   * @return  true if it should process the exception for the current event, false otherwise.
-   */
-  protected boolean acceptsEvent(MuleEvent event) {
-    return false;
+    return event.getError().getErrorType().equals(errorType);
```

Comment:
If we are now using the exact error type when parsing from the repository then the comment above can be deleted.